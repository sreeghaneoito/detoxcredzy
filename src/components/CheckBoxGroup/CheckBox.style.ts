import {StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  label: {
    fontSize: 16,
  },
  list: {
    marginTop: 20,
  },
  listItems: {
    marginTop: 8,
    height: 48,
    borderRadius: 4,
    alignItems: 'center',
    paddingLeft: 16,
    flexDirection: 'row',
  },
  Checkbox: {
    width: 18,
    height: 18,
    borderWidth: 1,
    borderColor: 'green',
    marginRight: 8,
  },
  rd4: {
    borderRadius: 4,
  },
  rd9: {
    borderRadius: 9,
  },
});
