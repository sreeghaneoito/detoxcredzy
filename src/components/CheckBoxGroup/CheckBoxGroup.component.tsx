import {View, Text, FlatList, Pressable} from 'react-native';
import React, {useState} from 'react';
import {globalStyle} from '../../styles/global.style';
import {style} from './CheckBox.style';
import Tick from '../../assets/icons/Tick';
import {useTheme} from '../../hooks/useTheme';
import {CheckBoxGroupProps} from '../../interfaces/component.interface';

/**
 * Checkbox Group
 * @param label: Checkbox group Heading
 * @param checkboxdata: An array of checkbox options
 * @param testid: Test id for automation
 * @returns JSX component
 */
const CheckBoxGroup = ({label, checkboxdata, testid}: CheckBoxGroupProps) => {
  const [data, setData] = useState(checkboxdata);
  const [nullSelected, setNullSelected] = useState(false);
  const {theme} = useTheme();
  const changeStatus = (indexDown: any) => {
    const newarr = data.map((obj, index) => {
      if (index === indexDown) {
        return {...obj, isSelected: !obj.isSelected};
      }
      return obj;
    });
    setData(newarr);
  };
  const nullOption = () => {
    setNullSelected(!nullSelected);
    const newarr = data.map(obj => {
      return {...obj, isSelected: false};
    });
    setData(newarr);
  };
  try {
    return (
      <View style={globalStyle.container} testID={testid}>
        <Text style={[style.label, theme.onBackground]}>{label}</Text>
        <FlatList
          style={style.list}
          data={data}
          renderItem={({item, index}) => {
            return (
              <Pressable
                style={[style.listItems, theme.surface]}
                onPress={() => {
                  changeStatus(index);
                }}
                testID={`checkboxitem${index}`}>
                <View style={[style.Checkbox, style.rd4]}>
                  {item.isSelected && <Tick />}
                </View>
                <Text style={theme.onSurface}>{item.name}</Text>
              </Pressable>
            );
          }}
          ListFooterComponent={() => {
            return (
              <Pressable
                style={[style.listItems, theme.surface]}
                onPress={() => {
                  nullOption();
                }}>
                <View style={[style.Checkbox, style.rd9]}>
                  {nullSelected && <Tick />}
                </View>
                <Text style={theme.onSurface}>
                  No, I donâ€™t have any of these.
                </Text>
              </Pressable>
            );
          }}
        />
      </View>
    );
  } catch (error) {
    console.log('Checkbox =>', error);
    return <View style={globalStyle.container} />;
  }
};

export default CheckBoxGroup;
