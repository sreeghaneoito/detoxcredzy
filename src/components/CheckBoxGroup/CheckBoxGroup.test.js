import React from 'react';
import renderer from 'react-test-renderer';
import CheckBoxGroup from './CheckBoxGroup.component';

test('renders correctly', () => {
  const checkboxgroup = renderer.create(<CheckBoxGroup />).toJSON();
  expect(checkboxgroup).toMatchSnapshot();
});
