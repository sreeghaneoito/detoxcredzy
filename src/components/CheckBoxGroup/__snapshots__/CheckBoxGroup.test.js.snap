// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  />
  <RCTScrollView
    ListFooterComponent={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 20,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "flexDirection": "row",
                "height": 48,
                "marginTop": 8,
                "paddingLeft": 16,
              },
              Object {
                "backgroundColor": "#F8F8FA",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "green",
                  "borderWidth": 1,
                  "height": 18,
                  "marginRight": 8,
                  "width": 18,
                },
                Object {
                  "borderRadius": 9,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            No, I donâ€™t have any of these.
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
